src/m365/pp/commands/dataverse/dataverse-table-list.ts

```diff
@@ -0,0 +1,81 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { powerPlatform } from '../../../../utils/powerPlatform';
+import PowerPlatformCommand from '../../../base/PowerPlatformCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  environment: string;
+  asAdmin: boolean;
+}
+
+class PpDataverseTableListCommand extends PowerPlatformCommand {
+  public get name(): string {
+    return commands.DATAVERSE_TABLE_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists dataverse tables in a given environment';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['SchemaName', 'EntitySetName', 'IsManaged'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environment <environment>'
+      },
+      {
+        option: '-a, --asAdmin'
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: any): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving list of tables for which the user is an admin...`);
+    }
+
+    const dynamicsApiUrl = await powerPlatform.getDynamicsInstanceApiUrl(args.options.environment, args.options.asAdmin);
```

Comment:
This function might be throwing an `Error`, it should be included in the `try catch` scope.

Reply:
@martinlingstuyl would the error noit be handled in the `handleError` that is part of our `Command` base class?